app/models/cloud_volume.rb

```diff
@@ -13,6 +13,10 @@ class CloudVolume < ApplicationRecord
   belongs_to :availability_zone
   belongs_to :cloud_tenant
   belongs_to :base_snapshot, :class_name => 'CloudVolumeSnapshot', :foreign_key => :cloud_volume_snapshot_id
+  belongs_to :storage_resource, :foreign_key => :storage_resource_id,
+             :class_name => "StorageResource", :inverse_of => :cloud_volumes
```

Comment:
If the association name matches the class name you don't have to declare `:class_name` or `:foreign_key` explicitly.  E.g `belongs_to :availability_zone` will default to `:class_name => "AvailabilityZone", :foreign_key => :availability_zone_id`